### PR TITLE
Update Django to 3.2.20

### PR DESCRIPTION
**Pull request by Jurppe**
* Update main Django core dependency from `3.2.13` to `3.2.20`.
  * Jira ticket of Snyk finding: https://helsinkisolutionoffice.atlassian.net/browse/KER-224?atlOrigin=eyJpIjoiNDhiNGExNDg1MDE3NDhmZTgyYTlkYTk0OTUzZDQyNWUiLCJwIjoiamlyYS1zbGFjay1pbnQifQ
  * CVE code: [CVE-2022-34265](https://www.cve.org/CVERecord?id=CVE-2022-34265)
* ~Also needed to downgrade `pyyml` from `5.4.1` to `5.3.1`~ because of installation error described here: https://github.com/yaml/pyyaml/issues/724
  * lets try `6.0.1 `instead because of the [CVE-2020-14343](https://www.cve.org/CVERecord?id=CVE-2020-14343) that `pyyaml 5.3.1` introduced

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Django>3.2<4
+Django>=3.2.14<4
```
Comment (danipran) [inline requirements.in:1]: Why 3.2.14 specifically? Patch releases should virtually never break anything.

**Review by danipran — COMMENTED** [review]
Not quite sure why you're restricting to 3.2.14 specifically instead of using the latest patch release (I think it's 3.2.20 atm)

File: requirements.in (hunk 1 above)
Comment (Jurppe) [reply requirements.in]: Valid point 👍

File: requirements.in [hunk 2]
```diff
@@ -30,3 +30,4 @@ sentry_sdk
 whitenoise
 social-auth-app-django>=5.0.0
 python-pptx
+pyyaml>=6
```
Comment (charn) [inline requirements.in:33]: This line is not actually required and I think it shouldn't be here unless it's a core requirement used in our own code. It helps when updating the requirement, but to bump a requirement you can also just run `pip-compile -P pyyaml requirements.in`

File: requirements.txt [hunk 3]
```diff
@@ -186,8 +188,6 @@ social-auth-core==4.1.0
     # via social-auth-app-django
 sqlparse==0.4.2
     # via django
-typing-extensions==4.2.0
```
Comment (charn) [inline requirements.txt:189]: You might actually need this if you run `pip-compile` inside the container with Python 3.7.

**Comment by charn** [comment]
Part of #417

**Pull request closed without merging**